modules/actions/pool/update-on-chain-data-v3.ts

```diff
@@ -0,0 +1,45 @@
+import { prisma } from "../../../prisma/prisma-client";
+import { networkContext } from "../../network/network-context.service";
+import { SUPPORTED_POOL_TYPES } from "../../pool/lib/pool-on-chain-data.service";
+import { Chain } from "@prisma/client";
+import { fetchTokenPairDataV3Router } from "../../pool/lib/pool-on-chain-tokenpair-data";
+import { prismaBulkExecuteOperations } from "../../../prisma/prisma-util";
+
+
+//TODO: Move this job to pool-on-chain-data-service with the necessary adaptations
+export async function updateOnChainDataV3(chain: Chain){
+  const balancerRouterAddress = networkContext.data.balancer.v3.routerAddress;
+  const filteredPools = await prisma.prismaPool.findMany({
+    where: {
+      vaultVersion: 3,
+      chain,
+      type: { in: SUPPORTED_POOL_TYPES },
```

Comment:
is the needed? if so, maybe a different name, so it's clear that it's related to types supporting the token contract function call used in the action?